### PR TITLE
Test for checking if FIN is generated after connection close

### PR DESCRIPTION
**Pull request by shivrai**
Test for checking whether **FIN** is generated when we close the connection after **read()**ing what the client sent. This behavior conforms to **RFC 793**.

### CODE REVIEW

File: gtests/net/packetdrill/tests/bsd/tcp/close/close-read-data-fin.pkt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// Client closes connection
+0.450 < F. 1001:1001(0) ack 1001 win 65535
+
+// TODO: ack or delayed ack ?
+0.460 > . 1001:1001(0) ack 1002 win 1026
+
+// We close the connection with a delay, in case there
+// is still data left to read
+0.470 close(4) = 0
+0.480 > F. 1001:1001(0) ack 1002 win 1026
```
Comment (tuexen) [inline gtests/net/packetdrill/tests/bsd/tcp/close/close-read-data-fin.pkt]: I don't think the kernel needs 10 ms to send out the FIN.

File: gtests/net/packetdrill/tests/bsd/tcp/close/close-read-data-fin.pkt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+// The success of the tests depends on the correctness
+// of the inbound sniffed packet on line 36.
```
Comment (tuexen) [inline gtests/net/packetdrill/tests/bsd/tcp/close/close-read-data-fin.pkt]: Line number is wrong.

**Comment by shivrai** [comment]
Fixed. Sorry had to do a force push.

File: gtests/net/packetdrill/tests/bsd/tcp/close/close-read-data-fin.pkt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// TODO: ack or delayed ack ?
+0.460 > . 1001:1001(0) ack 1002 win 1026
+
+// We close the connection with a delay, in case there
+// is still data left to read
+0.470 close(4) = 0
+0.470 > F. 1001:1001(0) ack 1002 win 1026
+
+// The success of the tests depends on the correctness
+// of the inbound sniffed packet on line 39.
```
Comment (tuexen) [inline gtests/net/packetdrill/tests/bsd/tcp/close/close-read-data-fin.pkt]: It is outbound, not inbound. Please fix the comment.

Comment (shivrai) [inline gtests/net/packetdrill/tests/bsd/tcp/close/close-read-data-fin.pkt]: Done.

**Pull request merged**